src/Builder.hs

```diff
@@ -139,8 +139,8 @@ builderProvenance = \case
     GhcCabal _ _     -> context Stage1 ghcCabal
     GhcPkg _ Stage0  -> Nothing
     GhcPkg _ _       -> context Stage0 ghcPkg
-    Haddock _        -> context Stage2 haddock
-    Hpc              -> context Stage1 hpcBin
+    Haddock _        -> context Stage1 haddock
```

Comment:
We shouldn't be changing the semantics of stages in this PR. I remember @angerman had some arguments for doing this and there were also some arguments against. We should do this separately and after a discussion in a separate issue, since it's not a trivial decision.